uk.co.compucorp.civicrm.tasksassignments/tasksassignments.php

```diff
@@ -12,6 +16,28 @@ function tasksassignments_civicrm_config(&$config) {
 }
 
 /**
+ * @param array $wrappers
+ * @param array $apiRequest
+ */
+function tasksassignments_civicrm_apiWrappers(&$wrappers, $apiRequest) {
+  $requestJson = ArrayHelper::value('json', $_REQUEST);
+  $params = ArrayHelper::value('params', $apiRequest, []);
+  $optionGroup = ArrayHelper::value('option_group_id', $params);
+  $isActivityType = $optionGroup === 'activity_type';
+  $entity = ArrayHelper::value('entity', $apiRequest);
+  $action = ArrayHelper::value('action', $apiRequest);
+
+  $jsonFromCaseEdit = '"actTypes":["OptionValue","get",{"option_group_id":"activity_type","options":{"sort":"name","limit":0}}]';
+  $isAssignmentEdit = strpos($requestJson, $jsonFromCaseEdit) !== FALSE;
+
+  if ($isAssignmentEdit && $entity === 'OptionValue' && $action === 'get' && $isActivityType) {
+    $wrappers[] = new StripInvalidTaskTypesWrapper();
+  } elseif ($entity === 'CaseType' && $action === 'create') {
```

Comment:
I would say that just hiding it is not enough. If we don't want to allow it then we should have validation to stop it. 

Also for some people who have already created assignments we should tell them if they already have activity types that are no valid.